services/director-v2/src/simcore_service_director_v2/api/routes/clusters.py

```diff
@@ -20,64 +37,187 @@
+    status_code=status.HTTP_201_CREATED,
+)
+async def create_cluster(
+    user_id: UserID,
+    new_cluster: ClusterCreate,
+    clusters_repo: ClustersRepository = Depends(get_repository(ClustersRepository)),
+):
+    return await clusters_repo.create_cluster(user_id, new_cluster)
+
+
+@router.get("", summary="Lists clusters for user", response_model=List[ClusterOut])
+async def list_clusters(
+    user_id: UserID,
+    clusters_repo: ClustersRepository = Depends(get_repository(ClustersRepository)),
+):
+    return await clusters_repo.list_clusters(user_id)
+
+
 @router.get(
     "/default",
-    summary="Returns the cluster details",
+    summary="Returns the default cluster",
     response_model=ClusterOut,
     status_code=status.HTTP_200_OK,
 )
 async def get_default_cluster(
     settings: DaskSchedulerSettings = Depends(get_scheduler_settings),
+):
+    assert settings.DASK_DEFAULT_CLUSTER_ID is not None  # nosec
+    raise HTTPException(
+        status_code=status.HTTP_501_NOT_IMPLEMENTED, detail="dev in progress"
+    )
+
+
+@router.get(
+    "/{cluster_id}",
+    summary="Get one cluster for user",
+    response_model=ClusterOut,
+    status_code=status.HTTP_200_OK,
+)
+async def get_cluster(
+    user_id: UserID,
+    cluster_id: ClusterID,
+    clusters_repo: ClustersRepository = Depends(get_repository(ClustersRepository)),
+):
+    try:
+        return await clusters_repo.get_cluster(user_id, cluster_id)
+    except ClusterNotFoundError as e:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"{e}") from e
+    except ClusterAccessForbiddenError as e:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=f"{e}") from e
+
+
+@router.patch(
+    "/{cluster_id}",
+    summary="Modify a cluster for user",
+    response_model=ClusterOut,
+    status_code=status.HTTP_200_OK,
+)
+async def update_cluster(
+    user_id: UserID,
+    cluster_id: ClusterID,
+    updated_cluster: ClusterPatch,
+    clusters_repo: ClustersRepository = Depends(get_repository(ClustersRepository)),
+):
+    try:
+        return await clusters_repo.update_cluster(user_id, cluster_id, updated_cluster)
+    except ClusterNotFoundError as e:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"{e}") from e
+    except ClusterAccessForbiddenError as e:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=f"{e}") from e
+    except ClusterInvalidOperationError as e:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@router.delete(
+    "/{cluster_id}",
+    summary="Remove a cluster for user",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def delete_cluster(
+    user_id: UserID,
+    cluster_id: ClusterID,
+    clusters_repo: ClustersRepository = Depends(get_repository(ClustersRepository)),
+):
+    try:
+        await clusters_repo.delete_cluster(user_id, cluster_id)
```

Comment:
actually I think it would make sense to remove the connection as well if exists. good catch.
also I guess I need to think of some timeout as well so as to not keep the connection alive for nothing.